### PR TITLE
:seedling: Bump k8s.io/* to v0.30.2 and controller-runtime to v0.18.4

### PR DESCRIPTION
**Pull request by mquhuy**
Bump multiple packages following CAPI main:
	k8s.io/api and k8s.io/apimachinery: v0.29.3 --> v0.30.2
	k8s.io/klog/v2: v2.110.1 --> v2.120.1
	sigs.k8s.io/controller-runtime: v0.17.3 --> v0.18.4
	k8s.io/client-go and k8s.io/component-base: v0.29.3 --> v0.30.2

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -1,16 +1,18 @@
 module github.com/metal3-io/ip-address-manager/api
 
-go 1.22
+go 1.22.0
```
Comment (tuminoid) [inline api/go.mod:3]:
```suggestion
go 1.22.3
```

File: api/go.mod [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
+
+toolchain go1.22.2
```
Comment (tuminoid) [inline api/go.mod:5]:
```suggestion
```

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,8 @@
 module github.com/metal3-io/ip-address-manager
 
-go 1.22
+go 1.22.0
```
Comment (tuminoid) [inline go.mod:3]:
```suggestion
go 1.22.3
```

File: go.mod [hunk 4]
```diff
@@ -1,6 +1,8 @@
...
+
+toolchain go1.22.2
```
Comment (tuminoid) [inline go.mod:5]:
```suggestion
```

File: hack/tools/go.mod [hunk 5]
```diff
@@ -1,10 +1,12 @@
 module github.com/metal3-io/ip-address-manager/hack/tools
 
-go 1.22
+go 1.22.0
```
Comment (tuminoid) [inline hack/tools/go.mod:3]: LGTM https://github.com/metal3-io/ip-address-manager/pull/562

it has all these hack/tools things done for you.

**Review by tuminoid — COMMENTED** [review]
Please retitle, and add more detailed description to your PR to reflect the content better, plus fix comments that remove the `toolchain` directive from go.mods.

Also LGTM https://github.com/metal3-io/ip-address-manager/pull/562 as it has all your hack/tools content already.

**Comment by tuminoid** [comment]
/retitle :seedling: Bump k8s.io/* to v0.30.2 and controller-runtime to v0.18.4

This is more descriptive thinking about the release log, as the CR and k8s version bump is the meaningful information, not individual component names.

**Review by tuminoid — COMMENTED** [review]
/approve

**Review by Sunnatillo — COMMENTED** [review]
/lgtm

**Pull request merged**